viewer/src/datamodels/cad/sector/CachedRepository.ts

```diff
@@ -63,154 +55,124 @@ export class CachedRepository implements Repository {
 
   async loadSector(sector: WantedSector): Promise<ConsumedSector> {
     const cacheKey = this.wantedSectorCacheKey(sector);
-    if (this._consumedSectorCache.has(cacheKey)) {
-      return this._consumedSectorCache.get(cacheKey);
-    }
-
-    switch (sector.levelOfDetail) {
-      case LevelOfDetail.Detailed: {
-        const loadOperation = this.loadDetailedSectorFromNetwork(sector).toPromise();
-        this._consumedSectorCache.forceInsert(cacheKey, loadOperation);
-        // Increase reference count to avoid geometry from being disposed
-        loadOperation.then(x => x.group?.reference());
-        return loadOperation;
+    try {
+      if (this._consumedSectorCache.has(cacheKey)) {
+        return this._consumedSectorCache.get(cacheKey);
       }
 
-      case LevelOfDetail.Simple: {
-        const loadOperation = this.loadSimpleSectorFromNetwork(sector).toPromise();
-        this._consumedSectorCache.forceInsert(cacheKey, loadOperation);
-        // Increase reference count to avoid geometry from being disposed
-        loadOperation.then(x => x.group?.reference());
-        return loadOperation;
-      }
+      switch (sector.levelOfDetail) {
+        case LevelOfDetail.Detailed: {
+          const consumed = await this.loadDetailedSectorFromNetwork(sector);
+          this._consumedSectorCache.forceInsert(cacheKey, consumed);
+          if (consumed.group !== undefined) {
+            // Increase reference count to avoid geometry from being disposed
+            consumed.group.reference();
+          }
+          return consumed;
+        }
 
-      case LevelOfDetail.Discarded:
-        return {
-          blobUrl: sector.blobUrl,
-          metadata: sector.metadata,
-          levelOfDetail: sector.levelOfDetail,
-          instancedMeshes: [],
-          group: undefined
-        };
+        case LevelOfDetail.Simple: {
+          const consumed = await this.loadSimpleSectorFromNetwork(sector);
+          this._consumedSectorCache.forceInsert(cacheKey, consumed);
+          // Increase reference count to avoid geometry from being disposed
+          if (consumed.group !== undefined) {
+            consumed.group.reference();
+          }
```

Comment:
I think you can do this as syntactic sugar
```suggestion
        consumed.group?.reference();
```